samples/nrf9160/modem_trace_flash/src/main.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/uart.h>
+#include <modem/lte_lc.h>
+#include <modem/nrf_modem_lib.h>
+#include <modem/nrf_modem_lib_trace.h>
+#include <nrf_modem.h>
+#include <nrf_modem_at.h>
+#include <dk_buttons_and_leds.h>
+
+#include "trace_flash.h"
+
+LOG_MODULE_REGISTER(modem_trace_flash_sample, CONFIG_MODEM_TRACE_FLASH_SAMPLE_LOG_LEVEL);
+
+#define UART1_DT_NODE DT_NODELABEL(uart1)
+
+static const struct device *const uart_dev = DEVICE_DT_GET(UART1_DT_NODE);
+
+/* define callback */
+LTE_LC_ON_CFUN(cfun_hook, on_cfun, NULL);
+
+/* callback implementation */
+static void on_cfun(enum lte_lc_func_mode mode, void *context)
+{
+	LOG_INF("LTE mode changed to %d\n", mode);
+}
+
+static void print_uart1(char *buf, int len)
+{
+	if (!device_is_ready(uart_dev)) {
+		LOG_ERR("uart1 device not found!");
+		return;
+	}
+	for (int i = 0; i < len; i++) {
+		uart_poll_out(uart_dev, buf[i]);
+	}
+}
+
+static void print_traces(void)
+{
+	uint8_t buf[1024];
+	int ret = sizeof(buf);
+	size_t read_offset = 0;
+
+	printk("Button 1 pressed - dumping traces to uart1\n");
+
+	/* Read out the trace data from flash */
+	while (ret == sizeof(buf)) {
+		ret = trace_flash_read(buf, sizeof(buf), read_offset);
+		if (ret < 0) {
+			LOG_ERR("Error reading modem traces: %d", ret);
+			break;
+		}
+		read_offset += ret;
+		print_uart1(buf, ret);
+	}
+	LOG_INF("Total trace bytes read from flash: %d", read_offset);
+}
+
+static void button_handler(uint32_t button_states, uint32_t has_changed)
+{
+	if (has_changed & button_states & DK_BTN1_MSK) {
+		print_traces();
+	}
+
+	if (has_changed & button_states & DK_BTN2_MSK) {
+		printk("\nButton 2 pressed - nothing here\n");
```

Comment:
```suggestion
		LOG_DBG("\nButton 2 pressed - nothing here\n");
```

Reply:
Think I'll just remove the button 2 handler. We can add it back when/if we need it.